doc/source/whatsnew/v0.19.0.txt

```diff
@@ -972,3 +972,5 @@ Bug Fixes
 
 - Bug in ``Index`` raises ``KeyError`` displaying incorrect column when column is not in the df and columns contains duplicate values (:issue:`13822`)
 - Bug in ``Period`` and ``PeriodIndex`` creating wrong dates when frequency has combined offset aliases (:issue:`13874`)
+
+- Bug where files opened by parsers (used by `pd.read_csv`, e.g.) were not closed properly. (:issue:`13940`)
```

Comment:
Indeed, when getting an iterator (`chunksize` or `iterator` specified) we [do not close](https://github.com/agraboso/pandas/blob/509e2e628d8447198ef90bd63c8912e434beafa4/pandas/io/parsers.py#L399-L400) the file. When the iterator has been consumed and raises `StopIteration`, we [close](https://github.com/agraboso/pandas/blob/509e2e628d8447198ef90bd63c8912e434beafa4/pandas/io/parsers.py#L904-L906) the file. If a user requests an iterator but does not consume it fully, s/he must call [.close()](https://github.com/agraboso/pandas/blob/509e2e628d8447198ef90bd63c8912e434beafa4/pandas/io/parsers.py#L733-L734) on it &mdash; though we could define a `__del__` and take care of that.